### PR TITLE
Basic API routing intitialised

### PR DESCRIPTION
**Pull request by code-ry**
Basic API routes defined
Basic Mongoose schemas and Models defined

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+try {
+    const m = await mongoose.connect(process.env.ATLAS_DB_URL)
+    console.log(m.connection.readyState === 1 ? 'Mongoose connected!' : 'Mongoose failed to connect')
+}
+catch (err) {
+    console.log(err)
+}
+
+// Define Book schema
+const bookSchema = new mongoose.Schema({
```
Comment (Hanzhi11) [inline db.js:25]: add the below?
condition
location
language
img
genre
description

File: db.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    title: { type: String, required: true}, 
+    author: { type: String, required: true}
+})
+
+// Create book model based on schema
+const BookModel = mongoose.model('Book', bookSchema)
+
+// Define Appointment schema
+const appointmentSchema = new mongoose.Schema({
```
Comment (Hanzhi11) [inline db.js:43]: divide name to first name and last name?
inc_book and out-book use objectID from the book schema?
add time?
remove location as it contains in out-book?

File: db.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    email: { type: String, required: true},
+    password: { type: String, required: true},
+    status: { type: String, required: true}
+})
+
+// Create User model based on schema
+const UserModel = mongoose.model('User', userSchema)
+
+// Define Location schema
+const locationSchema = new mongoose.Schema({
```
Comment (Hanzhi11) [inline db.js:71]: add email?

File: db.js (hunk 1 above)
Comment (code-ry) [reply db.js:25]: amending

File: db.js (hunk 2 above)
Comment (code-ry) [reply db.js:43]: amended

**Pull request merged**